test/conftest.py

```diff
@@ -1,15 +1,40 @@
-
 import sys
 
+import py
+import pytest
+
+from myhdl.conversion import analyze, verify
+from myhdl.conversion._verify import _simulators
+
+xfail = pytest.mark.xfail
+
+all_sims = list(_simulators)
+
+if sys.version_info[0] > 2:
+    collect_ignore = ['conversion/toVerilog/test_not_supported_py2.py']
 
 def pytest_addoption(parser):
-    parser.addoption("--include-reference", action="store_true",
-                     help="Include the reference design cosim tests")
+    parser.addoption("--sim", action="store", choices=all_sims,
+                     help="HDL Simulator")
 
 
 def pytest_configure(config):
-    sys._called_from_test = True
+    sim = config.getoption('sim')
+    if sim is not None:
+        verify.simulator = analyze.simulator = sim
```

Comment:
The default simulator should be set here, if present, then it doesn't need to be set in each file.  Also, the check if the simulator is present.

A new decorator could be made so the "simok" check doesn't need to be made in each test, `@skipif_no_sim`